### DISCUSSION TITLE
[QUESTION] Force an alias

### DISCUSSION THREAD

**Discussion by aesteve** [slick/slick#2853] [Questions] [answered]
Is there a way in Slick to force an alias for a selected field.
The use-case is the following. 
For now I'm writing this:
```
TableQuery[Person].map { person => my_udf('something', person.firstName, person.lastName) }
```
And it generates: 
```
SELECT my_udf('something', FIRSTNAME, LASTNAME) FROM PERSON
```
Which is excellent (all I'm caring about is the generated SQL, I'm not executing it using JDBC).

The only thing is that I'd like to provide my own alias for the SQL as such: 
```
SELECT my_udf('something', FIRSTNAME, LASTNAME) AS the_result FROM PERSON
```

I couldn't find a way to do it. Is this possible?
I thought about mapping the result on a case class as defined [here](https://scala-slick.org/doc/3.2.0/userdefined.html#monomorphic-case-classes) or maybe even providing an `as` extension method on `Rep` that would modify the way the `Rep` is represented as String (or extend `Rep` maybe).

Is there any pointer I've missed? Thank you.

Note: I understand that my use case might be unusual, since I only care about the SQL statement as a String, and not executing it against a driver. So most users shouldn't worry about aliasing the fields they selected. But just in case you might have give me a hint on how to workaround that (writing and managing my own extension method totally would work for me for instance)

**Comment by nafg** [comment, marked as answer]
I don't think so. Contributions welcome ;)

**Reply by aesteve** [reply]
Thank you.
I'd be more than happy to contribute but as I mentioned at this point I'm not even sure how it can be implemented.

I'm entirely lost on what I should do, what I should change, where I even should start from?

What approach would you recommend? What should I look at? `Rep[T]`? `Node`? Any pointer?

**Comment by nafg** [comment]
I don't know the answer.

That said we could try to set up a time some time to try to hack on it together

**Reply by aesteve** [reply]
I'd be happy to.
When are you usually available for hacking things on Slick? (and which time zone?)

Just so that I can setup my agenda accordingly.
Thanks a lot for the proposal.

**Reply by nafg** [reply]
Are you on Discord?
